app/controllers/recipes_controller.rb

```diff
@@ -6,4 +7,29 @@ def index
   def show
     @recipe = Recipe.find(params[:id])
   end
+
+  def edit
+    @recipe = Recipe.find(params[:id])
+  end
+
+  def update
+    @recipe = Recipe.find(params[:id])
+    if @recipe.update(recipe_params)
+      flash[:success] = 'Recipe updated successfully'
+      redirect_to recipe_path
+    else
+      render :edit
+    end
+
+  rescue ActiveRecord::RecordNotUnique
```

Comment:
It should be catched with uniquennes validation http://guides.rubyonrails.org/active_record_validations.html#uniqueness